feyton/atlp-backend
# Staging

#### What does this PR do?
This PR Fixes:
- Test not running
- Remove unused import and declarations
- DB Conflicts
- CORS Error when tested on Heroku

This PR Add:
- Swagger testing and documentation
- Refactor the codes

#### How should this be manually tested?
- Clone the repository
- `git checkout staging`
- `git fetch origin`
- `npm install`
- `npm run dev`
**Note** configure the DB in  _.env file. The project HIGHLY DEPENDS ON IT. Check the required
**.env**
`ACCESS_TOKEN_SECRET=" access token to sign with jwt"`
`REFRESH_TOKEN_SECRET="refresh secret"`
`MONGO_DB_URL = "mongodb://127.0.0.1:27017/database-name"`
`TESTING_DB_URL=mongodb://127.0.0.1:27017/database-for-testing  //Mocking will be added next`
`API_BASE= /api/v1`
`NODE_ENV="dev"`
`JWT_EXPIRATION="Your Time is seconds"`
`JWT_REFRESH_EXPIRATION="Your time in seconds"`
`SERVER_URL=http://127.0.0.1:3500`
`SERVER_NAME="LOCAL HOST"`


#### Any background context you want to provide?
This PR solve many of mentioned issues @Jkadhuwa 
#### What are the relevant pivotal tracker stories?
User are authorized on DOCS page
#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/28519395/149795099-8c260e19-d923-4aae-928c-45b526762bc5.png)


#### Questions:
N/A
